Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,238 @@ public static class BalanceStats
+        }
+
+        private static decimal GetLargestBalance(string[] peopleBalances)
+        {
+            var maxBalance = 0m;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                var isBalanceValid = decimal.TryParse(peopleBalances[i], out var balance);
+
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var biggestLossEver = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+
+                if (peopleBalances.Length <= 2)
+                {
+                    continue;
+                }
+
+                var biggestLossForPerson = GetBiggestLossForPerson(peopleBalances);
+
+                if (biggestLossForPerson < biggestLossEver)
+                {
+                    biggestLossEver = biggestLossForPerson;
+                    name = peopleBalances[0];
+                }
+            }
+
+            if (name == "")
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            var biggestLossEverNoComma = RemoveComma($"{biggestLossEver:C0}").Replace("(", "-").Replace(")", "");
+            return $"{name} lost the most money. {biggestLossEverNoComma}.";
+        }
+
+        private static decimal GetBiggestLossForPerson(string[] peopleBalances)
+        {
+            var biggestLoss = decimal.MaxValue;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                for (var j = i + 1; j < peopleBalances.Length; j++)
+                {
+                    var isBeforeBalanceValid = decimal.TryParse(peopleBalances[i], out var beginAmount);
+                    var isAfterBalanceValid = decimal.TryParse(peopleBalances[j], out var endAmount);
+
+                    if (isBeforeBalanceValid && isAfterBalanceValid)
+                    {
+                        biggestLoss = GetLoss(biggestLoss, beginAmount, endAmount);
+                    }
+                }
+            }
+
+            return biggestLoss;
+        }
+
+        private static decimal GetLoss(decimal biggestLoss, decimal beginAmount, decimal endAmount)
+        {
+            var loss = endAmount - beginAmount;
+            if (loss < biggestLoss)
+            {
+                biggestLoss = loss;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'biggestLoss'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)